Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -115,6 +130,11 @@ public class Interactable :
         protected float rollOffTime = 0.25f;
         protected float rollOffTimer = 0.25f;
 
+        // handle near touch interaction
```

Comment:
I'm seeing some common code here between PressableButton and Interactable.

We need to ensure that the behavior of pressable button matches Interactable, so therefore any changes / improvements we make to pressable button will need to also be duplicated here, which will make this code quite hard to maintain and likely to diverge from pressable button.

This seems like a very good candidate for removing the code duplication, either by:

1) Instantiating the pressable button at runtime and forwarding / listening to events, so that the logic lives in pressable button.
2) Abstracting out the state machine / logic for PressableButton, and then using it both here and in pressable button.


My intuition is option 2 will pay off more in the end, since then we could re-use that code even for other game engines.

Reply:
removed pressable button code from Interactable